src/plone/restapi/services/users/add.py

```diff
@@ -264,9 +264,13 @@ def update_password(self, data):
         mt = getToolByName(self.context, "portal_membership")
         pwt = getToolByName(self.context, "portal_password_reset")
 
+        # TODO error type and error message depending on login per email or userid
         if target_user is None:
-            self.request.response.setStatus(404)
-            return
+            return self._error(
+                403,
```

Comment:
This would be a breaking change I guess. Maybe it makes sense to separate the custom error message from the change of the HTTP response code.

Reply:
@ksuess I am currently thinking about if I'd do a breaking change release. Is this something you plan to continue to work on?